src/Configuration/RawConfiguration/RawConfiguration.php

```diff
@@ -33,78 +33,77 @@
 
 declare(strict_types=1);
 
-namespace Infection\Config;
+namespace Infection\Configuration\RawConfiguration;
 
-use Infection\Config\Validator as ConfigValidator;
-use Infection\Json\JsonFile;
-use Infection\Locator\FileNotFound;
-use Infection\Locator\Locator;
-use function Safe\getcwd;
-use Symfony\Component\Filesystem\Filesystem;
+use function file_get_contents;
+use function is_file;
+use function is_readable;
+use Seld\JsonLint\JsonParser;
+use Seld\JsonLint\ParsingException;
+use stdClass;
 
 /**
  * @internal
  */
-final class ConfigCreatorFacade
+final class RawConfiguration
 {
-    /**
-     * @var ConfigValidator
-     */
-    private $configValidator;
-
-    /**
-     * @var Locator
-     */
-    private $locator;
+    private $path;
 
     /**
-     * @var Filesystem
+     * @var stdClass|null
      */
-    private $filesystem;
+    private $decodedContents;
 
-    public function __construct(Locator $locator, Filesystem $filesystem)
+    public function __construct(string $path)
     {
-        $this->locator = $locator;
-        $this->filesystem = $filesystem;
-
-        $this->configValidator = new Validator();
+        $this->path = $path;
     }
 
-    public function createConfig(?string $customConfigPath): InfectionConfig
+    public function getPath(): string
     {
-        try {
-            $infectionConfigFile = $this->locateConfigFile($customConfigPath);
+        return $this->path;
+    }
 
-            $content = (new JsonFile($infectionConfigFile))->decode();
+    /**
+     * @throws InvalidFile
+     */
+    public function getDecodedContents(): stdClass
+    {
+        $this->initDecodedContents();
 
-            $configLocation = \pathinfo($infectionConfigFile, PATHINFO_DIRNAME);
-        } catch (FileNotFound $e) {
-            // Generate an empty class to trigger `configure` command.
-            $content = new \stdClass();
+        return $this->decodedContents;
+    }
 
-            $configLocation = getcwd();
+    /**
+     * @throws InvalidFile
+     */
+    private function initDecodedContents(): void
+    {
+        if (null !== $this->decodedContents) {
+            return;
         }
 
-        $infectionConfig = new InfectionConfig($content, $this->filesystem, $configLocation);
-
-        $this->configValidator->validate($infectionConfig);
+        if (!is_file($this->path)) {
+            throw InvalidFile::createForFileNotFound($this);
+        }
 
-        return $infectionConfig;
-    }
+        if (!is_readable($this->path)) {
+            throw InvalidFile::createForFileNotReadable($this);
+        }
 
-    private function locateConfigFile(?string $customConfigPath): string
-    {
-        $configPaths = [];
+        $contents = @file_get_contents($this->path);
```

Comment:
Silencing errors probably isn't a good idea. Why not use the `Safe\file_get_contentes` function? Then we can remov all the earlier checks, and let that handle it for us.

Reply:
I think switching to `Safe\file_get_contents` is a good idea.

Removing the earlier checks however highly depends on how explicit the error messages are, will check.